cni/network/network.go

```diff
@@ -570,19 +575,28 @@ func (plugin *NetPlugin) Add(args *cniSkel.CmdArgs) error {
 	return nil
 }
 
+// cleanup allocated ipv4 and ipv6 addresses if they exist
 func (plugin *NetPlugin) cleanupAllocationOnError(
 	result, resultV6 *cniTypesCurr.Result,
 	nwCfg *cni.NetworkConfig,
 	args *cniSkel.CmdArgs,
 	options map[string]interface{},
 ) {
 	if result != nil && len(result.IPs) > 0 {
-		if er := plugin.ipamInvoker.Delete(&result.IPs[0].Address, nwCfg, args, options); er != nil {
+		addresses := make([]*net.IPNet, len(result.IPs))
+		for i, ip := range result.IPs {
```

Comment:
instead of doing this. we should keep same delete method signature and iterate over result.IPs and pass that

Reply:
btw why we change to array. are we expecting more than 1 ipv4/ipv6 addresses here?